### PR TITLE
Add csharp layer.

### PR DESCRIPTION
**Pull request by VyronLee**
# PR Prelude

Thank you for working on SpaceVim! :)

**Please complete these steps and check these boxes (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [x] I have read and understood SpaceVim's [CONTRIBUTING][cont] document.
- [x] I have read and understood SpaceVim's [CODE_OF_CONDUCT][code] document.
- [x] I have included tests for the changes in my PR. If not, I have included a
  rationale for why I haven't.
- [x] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful

[Please explain **in detail** why the changes in this PR are needed.]

Lack of language csharp layer.

[cont]: https://github.com/SpaceVim/SpaceVim/blob/master/CONTRIBUTING.md
[code]: https://github.com/SpaceVim/SpaceVim/blob/master/CODE_OF_CONDUCT.md

### CODE REVIEW

File: autoload/SpaceVim/layers/lang/csharp.vim [hunk 1]
```diff
@@ -0,0 +1,113 @@
+"=============================================================================
+" csharp.vim --- SpaceVim lang#csharp layer
+" Copyright (c) 2016-2018 VyronLee & Contributors
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/csharp.vim]: the copyright should same as other layers.

File: autoload/SpaceVim/layers/lang/csharp.vim [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+" Author: VyronLee < lwz_jz@hotmail.com >
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/csharp.vim]: you can change @ to # to avoid hack email

File: autoload/SpaceVim/layers/lang/csharp.vim [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+"   normal          SPC l d         show doc
+"   normal          SPC l g g       go to definition
+"   normal          SPC l g i       find implementations
+"   normal          SPC l g t       find type
+"   normal          SPC l g s       find symbols
+"   normal          SPC l g u       find usages of symbol under cursor
+"   normal          SPC l g m       find members in the current buffer
+"   normal          SPC l r r       rename symbol under cursor
+"   normal          SPC l s r       reload the solution
+"   normal          SPC l s s       start the OmniSharp server
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/csharp.vim]: I recommend to use same language specific key bindings as other language, then when user edit more then one languages, they do not need to remember key bindings.

Comment (VyronLee) [reply autoload/SpaceVim/layers/lang/csharp.vim]: ok, I will modify it and recommit later.

Comment (VyronLee) [reply autoload/SpaceVim/layers/lang/csharp.vim]: @wsdjeg How about the non suggested key bindings? Can I still use 2-level binding, such as ` SPC l s r   -> reload the solution`?

Comment (wsdjeg) [reply autoload/SpaceVim/layers/lang/csharp.vim]: OK

**Comment by VyronLee** [comment]
changes committed, review again, please.

File: autoload/SpaceVim/layers/lang/csharp.vim [hunk 4]
```diff
@@ -0,0 +1,109 @@
+"=============================================================================
+" csharp.vim --- SpaceVim lang#csharp layer
+" Copyright (c) 2016-2017 VyronLee & Contributors
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/csharp.vim]: I am sorry, as we are using `GPLv3` License, so the copyright should be 

```
" Copyright (c) 2016-2017 Wang Shidong & Contributors
```

you can use your info after `Author:`.

Comment (VyronLee) [reply autoload/SpaceVim/layers/lang/csharp.vim]: It's my fault.

**Comment by wsdjeg** [comment]
Thanks, merged.

**Pull request merged**